File: API/Helpers/UserParams.cs
using System;

namespace API.Helpers;

public class UserParams : PaginationParams
{
    public  string? Gender { get; set; }
    public string? CurrentUserName { get; set; }
    public string? City { get; set; }

    public string? SearchString { get; set; }
    public int Age { get; set; }
    public int MaxAge { get; set; } = 150;
    public int MinAge { get; set; } = 18;
    public string? OrderBy { get; set; } = "lastActive";
}

File: client/src/app/models/userParams.ts
import { User } from "./user";

export class UserParams{
  gender : string;
  minAge = 18;
  maxAge = 150;
  pageNumber = 1;
  pageSize = 12;
  orderBy = 'lastActive';
  searchString  = '';

  constructor(user : User | null){
    this.gender = ""
  }
}
